### PR TITLE
gc_onarc

### LINKED ISSUE #179: Great Circle - gc_onarc
`gc_onarc` NCL function as part of Great Circle functionality
- NCL entry for [gc_onarc](https://www.ncl.ucar.edu/Document/Functions/Built-in/gc_onarc.shtml)
- NCL receipt for `gc_onarc`

### PR DESCRIPTION
**Pull request by cyschneck**
## PR Summary
Covers `gc_onarc` to determine if a point on the globe lies on a specified great circle arc

## Related Tickets & Documents
Closes #179 

## PR Checklist
**General**
- [x] PR includes a summary of changes
- [x] Link relevant issues, make one if none exist
- If adding a new page, select which type:
  - [x] NCL Entry
  - [x] Receipt

### CODE REVIEW

**Comment by cyschneck** [comment]
Difficult to have wide testing, but the tests here are on based on edge cases from [uxarray.grid.arcs.point_within_gca](https://github.com/UXARRAY/uxarray/blob/main/test/test_arcs.py) testing

File: ncl/receipts/great_circle.ipynb [hunk 1]
```diff
@@ -468,6 +468,16 @@
...
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "295d1c1b-43ea-49ed-a6f2-514539eb4900",
+   "metadata": {},
+   "source": [
+    "<div class=\"admonition alert alert-info\">\n",
+    "    <p class=\"admonition-title\" style=\"font-weight:bold\">Important Note</p>\n",
+    "    The tolerance for the function to check if a point lies on a great circle arc varies between NCL (`1e-10`) and UXarray (`1e-8`) which can generate some potential mismatches when comparing between each function"
```
Comment (kafitzgerald) [inline ncl/receipts/great_circle.ipynb:478]: Doesn't UXarray use `np.isclose()` for this with both an absolute and relative tolerance?  And I think NCL is using degrees and UXarray cartesian coordinates so there's some additional complexity with that as well.     

It might be easier (and more robust) to just mention that they're different and refer folks to the docs/code.  Maybe not quite as helpful, but...

Comment (cyschneck) [reply ncl/receipts/great_circle.ipynb:478]: I can update it, from the user's perspective it just returns a boolean
